### PR TITLE
Chore(sdk): Adds SDK Template for Cloud based experiments

### PR DESCRIPTION
**Pull request by uditgaurav**
Signed-off-by: Udit Gaurav <udit.gaurav@harness.io>

**What this PR does / why we need it**:

- Adds SDK support for cloud based experiments like AWS, GCP and Azure
- The template will also be good to work with VMWare experiments

New Templates:

- [x] AWS
- [x] VMWare
- [x] Azure
- [x] GCP

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
-   [ ] Fixes #<issue number>
-   [ ] PR messages has document related information
-   [ ] Labelled this PR & related issue with `breaking-changes` tag
-   [ ] PR messages has breaking changes related information
-   [ ] Labelled this PR & related issue with `requires-upgrade` tag
-   [ ] PR messages has upgrade related information
-   [ ] Commit has unit tests
-   [ ] Commit has integration tests
-   [ ] E2E run Required for the changes

### CODE REVIEW

File: contribute/developer-guide/README.md [hunk 1]
```diff
@@ -113,6 +113,11 @@ The *generate_experiment.go* script is a simple way to bootstrap your experiment
       - Provide the type of chaoslib in the `-t` flag. It supports the following values:
            - `exec`: It creates the exec based chaoslib(default type)
            - `helper`: It creates the helper based chaoslib
+           - `aws`: It creates experiment template for aws platform.
```
Comment (ispeakc0de) [inline contribute/developer-guide/README.md:116]:
```suggestion
           - `aws`: It creates sample experiment for aws platform.
```

Comment (ispeakc0de) [reply contribute/developer-guide/README.md:116]: update the same at other places

File: contribute/developer-guide/templates/chaoslib_non-k8s.tmpl [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+				common.SetTargets(id, "injected", "TARGET", chaosDetails)
+			}
+
+			for _, id := range targetIDList {
+
+				// @TODO: user WAIT-FOR-CHAOS-INJECTION
+				// WAIT UNTIL THE CHAOS IS INJECTED COMPLETLY 
+
+                // @TODO: REPLACE THE TARGET WITH THE SERVICE UNDER CHAOS
+				common.SetTargets(id, "reverted", "TARGET", chaosDetails)
```
Comment (ispeakc0de) [inline contribute/developer-guide/templates/chaoslib_non-k8s.tmpl:178]: reverted should be in end - after chaos reverted successfully

**Review by Jonsy13 — APPROVED** [review]
Looks good, PTAL on comments provided by @ispeakc0de

**Review by ksatchit — APPROVED** [review]

**Review by ispeakc0de — APPROVED** [review]

**Pull request merged**